include/exadg/incompressible_navier_stokes/spatial_discretization/operator_coupled.cpp

```diff
@@ -258,8 +250,9 @@ OperatorCoupled<dim, Number>::solve_nonlinear_problem(BlockVectorType &  dst,
   // Update nonlinear operator
   nonlinear_operator.update(rhs_vector, time, scaling_factor_mass);
 
-  // Update linear operator
```

Comment:
I do not get line 262 in the old code - is `linear_operator` the *linearized* operator? Why was the `momentum_operator` not updatedpreviously? Is that the bug you were mentioning before?

Reply:
The linear operator is just needed as the class handed over to the Krylov solver as the matrix or operator, providing the function `vmult()`. For our matrix-free implementation, `momentum_operator` is the relevant spatial operator.

The problem or potential bug was that `linear_operator->update()` did not immediately update the `momentum_operator`, which however would have been necessary for a correct update of the preconditioner.